test/integration/consul-container/test/traffic_permissions/tcp_test.go

```diff
@@ -0,0 +1,638 @@
+								Namespace:    "default",
+								Partition:    "default",
+								Peer:         "local",
+							},
+						},
+					},
+				},
+			},
+			tp2: &pbauth.TrafficPermissions{
+				Destination: &pbauth.Destination{
+					IdentityName: staticServerIdentity,
+				},
+				Action: pbauth.Action_ACTION_ALLOW,
+				Permissions: []*pbauth.Permission{
+					{
+						Sources: []*pbauth.Source{
+							{
+								IdentityName: "static-client-1-identity",
+								Namespace:    "default",
+								Partition:    "default",
+								Peer:         "local",
+							},
+						},
+					},
+				},
+			},
+			client1TCPSuccess:  false,
+			client1EchoSuccess: false,
+			client2TCPSuccess:  false,
+			client2EchoSuccess: false,
+		},
+		"deny all exclude service + allow on that service": {
+			tp1: &pbauth.TrafficPermissions{
+				Destination: &pbauth.Destination{
+					IdentityName: staticServerIdentity,
+				},
+				Action: pbauth.Action_ACTION_DENY,
+				Permissions: []*pbauth.Permission{
+					{
+						Sources: []*pbauth.Source{
+							{
+								Namespace: "default",
+								Partition: "default",
+								Peer:      "local",
+								Exclude: []*pbauth.ExcludeSource{
+									{
+										IdentityName: "static-client-1-identity",
+										Namespace:    "default",
+										Partition:    "default",
+										Peer:         "local",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tp2: &pbauth.TrafficPermissions{
+				Destination: &pbauth.Destination{
+					IdentityName: staticServerIdentity,
+				},
+				Action: pbauth.Action_ACTION_ALLOW,
+				Permissions: []*pbauth.Permission{
+					{
+						Sources: []*pbauth.Source{
+							{
+								IdentityName: "static-client-1-identity",
+								Namespace:    "default",
+								Partition:    "default",
+								Peer:         "local",
+							},
+						},
+					},
+				},
+			},
+			client1TCPSuccess:  true,
+			client1EchoSuccess: true,
+			client2TCPSuccess:  false,
+			client2EchoSuccess: false,
+		},
+	}
+
+	runTrafficPermissionsTests(t, true, cases)
+}
+
+func TestTrafficPermission_TCP_DefaultAllow(t *testing.T) {
+	cases := map[string]trafficPermissionsCase{
+		"default allow": {
+			tp1:                nil,
+			client1TCPSuccess:  true,
+			client1EchoSuccess: true,
+			client2TCPSuccess:  true,
+			client2EchoSuccess: true,
+		},
+		"allow everything": {
+			tp1: &pbauth.TrafficPermissions{
+				Destination: &pbauth.Destination{
+					IdentityName: staticServerIdentity,
+				},
+				Action: pbauth.Action_ACTION_ALLOW,
+				Permissions: []*pbauth.Permission{
+					{
+						Sources: []*pbauth.Source{
+							{
+								Namespace: "default",
+								Partition: "default",
+								Peer:      "local",
+							},
+						},
+					},
+				},
+			},
+			client1TCPSuccess:  true,
+			client1EchoSuccess: true,
+			client2TCPSuccess:  true,
+			client2EchoSuccess: true,
+		},
+		// TODO I don't like this behavior.
+		"allow one protocol doesnt impact the other protocol": {
```

Comment:
@ishustava It feels weird to me that configuring a traffic permission with default allow only swaps to default deny for a single port rather than all port. Should we change this behavior to swap all ports to default deny?